workflow/proc_pipe/tractoflow/run_tractoflow.py

```diff
@@ -0,0 +1,460 @@
+            
+    else:
+        
+        logger.info("Continue checks assuming 1 directed file...")
+
+        ## pull the index of the bvec that exists
+        didx = np.argmax(cbv)
+        
+        ## pull phase encoding for directed volume
+        fpe = cpe[didx]
+
+        ## clean fpe of unnecessary + if it's there
+        if fpe[-1] == "+":
+            fpe = fpe[0]
+        
+        ## determine the reverse phase encoding
+        if (len(fpe) == 1):
+            rpe = fpe + "-"
+        else:
+            rpe = fpe[0]
+
+        logger.info(f"Foward Phase Encoding:  {fpe}")
+        logger.info(f"Reverse Phase Encoding: {rpe}")
+            
+        ## look for the reverse phase encoded file in the other candidates
+        if (rpe in cpe):
+            
+            rpeb0 = dmri_files[cpe.index(rpe)]
+            rpevol = rpeb0.get_image()
+            tmp_dir = tempfile.mkdtemp() #TemporaryDirectory()
+            rpe_out = f'{participant_id}_rpe_b0.nii.gz'
+            
+            ## if the rpe file has multiple volumes
+            if len(rpevol.shape) == 4:
+
+                logger.info('An RPE file is present: Averaging b0 volumes to single RPE volume...')
+                ## load and average the volumes
+                rpedat = rpevol.get_fdata()
+                rpedat = np.mean(rpedat, 3)
+
+                ## and write the file to /tmp
+                rpe_data = nib.nifti1.Nifti1Image(rpedat, rpevol.affine)
+                rpe_shape = rpe_data.shape
+                nib.save(rpe_data, Path(tmp_dir, rpe_out).joinpath())
+                
+            else:
+
+                logger.info('An RPE file is present: Copying single the single RPE b0 volume...')
+                ## otherwise, copy the input file to tmp
+                rpe_shape = rpevol.shape
+                shutil.copyfile(rpeb0, rpe_out)
+                            
+        else:
+            
+            logger.info("No RPE is found in candidate files")
+            rpe_out = None
+            
+    logger.info("= "*25)
+    
+    ## default assignments
+    dmrifile = Path(bids_dir, participant_id, 'ses-' + session_id, 'dwi', dmri_files[didx].filename).joinpath()
+    bvalfile = Path(bids_dir, participant_id, 'ses-' + session_id, 'dwi', dmri_files[didx].filename.replace('.nii.gz', '.bval')).joinpath()
+    bvecfile = Path(bids_dir, participant_id, 'ses-' + session_id, 'dwi', dmri_files[didx].filename.replace('.nii.gz', '.bvec')).joinpath()
+    anatfile = Path(bids_dir, participant_id, 'ses-' + session_id, 'anat', oanat.filename).joinpath()
+
+    ## condition empty path
+    if rpe_out == None:
+        rpe_file = None
+    else:
+        rpe_file = Path(tmp_dir, rpe_out).joinpath()
+        
+    ## return the paths to the input files to copy
+    return(dmrifile, bvalfile, bvecfile, anatfile, rpe_file)
+
+def run_tractoflow(participant_id, global_configs, session_id, output_dir, use_bids_filter, logger=None):
```

Comment:
Could we rename this to `run` - i think with `run_mr_proc.py` or `DaskQueue` calling these things in future, we may want standardized naming for the main "run" function that gets called externally.